### DISCUSSION TITLE
Flywoo f745Nano No USB after flash.

### DISCUSSION THREAD

**Discussion by davidabevan** [iNavFlight/inav#7546] [Q&A]
I cant seem to resolve this. tried two laptops also stm drivers re install.
I have the Flywoo explorer Lr4 v2 Hd running the Caddx Vista, 
Problem.
Runs fine on betaflight. If I flash Inav I cant connect, The laptop device manager sees the stm32 comm port.
I tried all the Flywoo targets on their web page only one works for me however motor 3 not active on this target, Its listed as the as the inav_3.0.0_FLYWOO_F745HD.hex   
https://cdn.shopifycdn.net/s/files/1/0010/7410/2324/files/inav_3.0.0_FLYWOO_F745HD.hex?v=1629798243
So I tried to recompile from source, different laptops drivers usb leads removing the bluetooth module powering up and connecting. I have run out of ideas and was trying to compare to the betaflight config, it seems the same. 
Can the vcp be mapped wrong, If so how can i Modify.
Any Ideas really appreciated I have been on and off trying to get Inav working on this board since March

**Comment by matstey** [comment]
Did u get anywhere with this? I am experiencing the exact same issue but with the F745 HEX board. Flywoo just point me to the builds on their site.

**Comment by davidabevan** [comment]
Yes built my own target. I haven't details at hand nor am I an expert by
any means at all in programming but if you are able to compile from source
I can post my alterations here. Bear in mind this is only tested on my f745
nano
Regards Dave B

**Comment by davidabevan** [comment]
Just a thought I've got my binarys I can upload that if you can't compile
from source
Dave

**Comment by matstey** [comment]
Oh great. The changes would be great then i could submit a PR hopefully.

**Comment by davidabevan** [comment]
Hi Mat.
Here is my code, Please include my name in any public release as I spent a
long time getting this working as I was not getting any help from community
regarding the issues I had tring to work it out. I also was in contact with
***@***.*** they have had my code a long time so obviously didn't publish
it, Ive flown inav 4.1 with this for 6 months no issues as far as i am
concerned board fully works including  leds.
Here is my 4.1 hex for f745 as well
Regards
Dave Bevan

On Wed, 8 Jun 2022 at 21:58, Mat Stevenson ***@***.***> wrote:

> Oh great. The changes would be great then i could submit a PR hopefully.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/iNavFlight/inav/discussions/7546#discussioncomment-2908819>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIFZFZPEZJABGM4W6OAPLDDVOECOLANCNFSM5YHYIQKQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

target_stm32f745xg(FLYWOOF745NANO)

**Comment by Mat-Stevenson-Simworx** [comment]
Thanks. I appreciate your effort on this. I will try my best to get a PR approved for this and make sure your are credited. The hex and code doesnt seem to have come though. I dont think github supports attachments when replying as an email. You should beable to attach the files if you reply on github directly.

**Reply by davidabevan** [reply]
will try again....

**Comment by davidabevan** [comment]
here we go Mat. Fingers crossed.
[FLYWOOF745.zip](https://github.com/iNavFlight/inav/files/8868171/FLYWOOF745.zip)

**Reply by Mat-Stevenson-Simworx** [reply]
Perfect. Thanks. I ll let you know once i v tested and got the PR up.

**Comment by davidabevan** [comment]
Yes I hope it helps this was so frustrating as was a perfect combo for inav
thats what drove me to keep digging away at the code till i solved it.
Hopefully it can help others try this little board out.
Dave

**Comment by matstey** [comment]
Dont suppose your board came on a digital bnf and has what i am assuming to be the osd chip missing
![goku_f745nano_compare](https://user-images.githubusercontent.com/44748411/172948227-6f602ec7-2884-449b-b5ff-5084cc3e8a45.png)
?

**Reply by davidabevan** [reply]
left one Is mine. The binary i uploaded should work

**Reply by davidabevan** [reply]
I in fact think this is the issue with the working and non working usb connect. I think the published  code may be for the right hand fc there. this board on the left does not use the 2nd i2c bus and thats what caused my usb fail. the other work was mainly motor re order rather than as the pin map to esc is not soldered but hard wired

**Comment by davidabevan** [comment]
Indeed it does the lr 4 with caddx vista
